bulbs/content/tasks.py

```diff
@@ -51,3 +63,117 @@ def update_feature_type_rates(featuretype_pk):
                 rate=0,
                 feature_type_id=featuretype_pk,
                 role_id=role.pk)
+
+
+@shared_task(default_retry_delay=5)
+def post_to_instant_articles_api(content_pk):
+    from .models import Content
+
+    if getattr(settings, 'FACEBOOK_API_ENV', '').lower() == 'production':
+        logger = logging.getLogger(__name__)
+
+        content = Content.objects.get(pk=content_pk)
+        feature_type = getattr(content, 'feature_type', None)
+
+        fb_page_id = getattr(settings, 'FACEBOOK_PAGE_ID', '')
+        fb_api_url = getattr(settings, 'FACEBOOK_API_BASE_URL', '')
+        fb_access_token = vault.read(settings.FACEBOOK_TOKEN_VAULT_PATH)
+
+        # if feature type is IA approved & content is published
+        if feature_type and feature_type.instant_article and content.is_published:
+            # render page source
+            context = {
+                'content': content,
+                'absolute_uri': getattr(settings, 'WWW_URL'),
+                'transformed_body': transform(
+                    getattr(content, 'body', ''),
+                    InstantArticleRenderer())
+            }
+            try:
+                source = render_to_string(
+                    'instant_article/_instant_article.html', context
+                )
+            except TemplateDoesNotExist:
+                source = render_to_string(
+                    'instant_article/base_instant_article.html', context
+                )
+
+            # Post article to instant article API
+            post = requests.post(
+                '{0}/{1}/instant_articles'.format(fb_api_url, fb_page_id),
+                data={
+                    'access_token': fb_access_token,
+                    'html_source': source,
+                    'published': 'true',
+                    'development_mode': 'false'
+                })
+
+            if not post.ok:
+                logger.error('''
+                    Error in posting Instant Article.\n
+                    Content ID: {0}\n
+                    IA ID: {1}\n
+                    Status Code: {2}'''.format(
+                        content.id,
+                        content.instant_article_id,
+                        post.status_code))
+                return
+
+            # Poll for status of article
+            response = ""
+            while response != "SUCCESS":
```

Comment:
Possible infinite loop if response never contains "success" or "error"  status -- any concern FB service might get stuck and keep returning a pending state? I'd include some kind of polling timeout to error out after say 5 minutes.